packages/apps/reputation-oracle/server/src/modules/auth/strategy/jwt.http.ts

```diff
@@ -59,6 +62,27 @@ export class JwtHttpStrategy extends PassportStrategy(Strategy, 'jwt-http') {
       );
     }
 
+    const jwt = getJwtFromBearer(request.headers.authorization);
+
+    if (!jwt) {
+      throw new ControlledError(
+        'User has invalid jwt data',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
+
+    const jwtData = await this.jwtService.decode(jwt);
+
+    if (
+      (user.evmAddress && user.evmAddress !== jwtData.address) ||
+      (!user.evmAddress && jwtData.address)
+    ) {
+      throw new ControlledError(
+        'User has invalid jwt data',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
+
```

Comment:
Why do we need this?

Reply:
If an address is associated with a user, it must match the address that is in the JWT.

Reply:
If address in JWT payload doesn't match with the address in a db it means that JWT was malformed which means that JWT validation will fail.

Reply:
Adding to what @Dzeranov said, there are 2 options for sign in:
1. Operators sign in via web3, it means the address is verified when the JWT is generated, so in this case it won't be necessary.
2. Workers sign in via user and pass, but those have to link the evm address to their account signing a message. After that, for now, the address cannot be changed, so the possibility of using a wrong address I would say is 0. For this to happen, we should manually update the evm address in the database.

Reply:
Yes, I have rechecked how `registerAddress` works, you are right.
Implementation was not correct. I fixed it.